apps/framework-cli/src/framework/consumption/loader.rs

```diff
@@ -1,50 +1,90 @@
+use super::model::{Consumption, ConsumptionQueryParam, EndpointFile};
+use crate::framework::languages::SupportedLanguages;
+use crate::framework::python::executor::{run_python_program, PythonProgram};
 use crate::project::Project;
-use sha2::{Digest, Sha256};
-use std::{fs, io, path::Path};
-
-use super::model::{Consumption, EndpointFile};
 use crate::utilities::PathExt;
+use serde::Deserialize;
+use sha2::{Digest, Sha256};
+use std::{fs, path::Path};
 
 #[derive(Debug, thiserror::Error)]
 #[non_exhaustive]
 pub enum ConsumptionLoaderError {
     #[error("Failed to open file: {0}")]
-    FailedToOpenFile(#[from] std::io::Error),
+    FailedToOpenFile(std::io::Error),
+    #[error("Failed to load query params: {0}")]
+    FailedToLoadParams(std::io::Error),
 }
 
-pub fn load_consumption(project: &Project) -> Result<Consumption, ConsumptionLoaderError> {
-    let endpoint_files = walkdir::WalkDir::new(project.consumption_dir())
+#[derive(Debug, Deserialize)]
+struct QueryParamOutput {
+    params: Vec<ConsumptionQueryParam>,
+}
+
+pub async fn load_consumption(project: &Project) -> Result<Consumption, ConsumptionLoaderError> {
+    let mut endpoint_files = Vec::new();
+    for f in walkdir::WalkDir::new(project.consumption_dir())
         .into_iter()
-        .filter_map::<EndpointFile, _>(|f| {
-            if let Ok(f) = f {
-                if f.file_type().is_file() && f.path().ext_is_supported_lang() {
-                    build_endpoint_file(project, f.path()).ok().flatten()
-                } else {
-                    None
-                }
-            } else {
-                None
+        // drop Err cases
+        .flatten()
+    {
+        if f.file_type().is_file() && f.path().ext_is_supported_lang() {
+            let result = build_endpoint_file(project, f.path()).await;
+            log::debug!("build_endpoint_file result: {:?}", result);
+            if let Some(file) = result.ok().flatten() {
+                endpoint_files.push(file);
             }
-        })
-        .collect();
+        }
+    }
 
     Ok(Consumption { endpoint_files })
 }
 
-fn build_endpoint_file(
+async fn load_python_query_param(
```

Comment:
Can we not have `python` specific code that is not in the python module?
I would like to keep the separation clean